git.py

```diff
@@ -85,10 +86,17 @@ def shallow_clone(self, depth=1):
 
     def amend_commit(self):
         return self._git.commit('--amend', '--no-edit')
-
+        
     def apply_patch(self, patch: Patch):
         print(patch.text_with_headers)
-        return self._git.am(patch.text_with_headers)
+        try:
+            output = self._git.am(patch.text_with_headers)
+            return output
+        except subprocess.CalledProcessError:
+            print('Failed to amend. Aborting...')
```

Comment:
nit: s/ammend/apply patch.
> git-am - Apply a series of patches from a mailbox

Reminder you still need to make the change for `git()` to error out when the cmd fails via `check_call` or similar.

Reply:
does check = true in the run part cover that?

Reply:
Yes, `check=True` should be sufficient to cause it to raise the Exception we expect.